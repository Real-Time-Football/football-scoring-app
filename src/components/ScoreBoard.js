import React from 'react'
import TeamScore from './TeamScore'
import styled, { keyframes } from 'styled-components'

const ScoreContainer = styled.article`
    display: flex;
    justify-content: center;
    align-items: center;
    font-family: Arial, Helvetica, sans-serif;
    padding: 30px 0;
`

const ErrorContainer = styled.div`
    font-family: Arial, Helvetica, sans-serif;
    color: #721c24;
    background-color: #f8d7da;
    position: relative;
    padding: 15px 20px;
    margin-bottom: 10px;
    border: 1px solid #f5c6cb;
    border-radius: 5px;
`

const dualRingRotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`

const LoadingDualRing = styled.div`
    display: inline-block;
    width: 80px;
    height: 80px;

    &:after {
        content: ' ';
        display: block;
        width: 32px;
        height: 32px;
        margin: 10px;
        border-radius: 50%;
        border: 3px solid #6495ed;
        border-color: #6495ed transparent #6495ed transparent;
        animation: ${dualRingRotate} 1.2s linear infinite;
    }
`

class ScoreBoard extends React.PureComponent {
    state = {
        error: null,
        isLoaded: false,
        scoreHome: 0,
        scoreVisitors: 0,
    }

    componentDidMount() {
        fetch('http://localhost:8080/match/4e9ea4e7-007e-46c0-b738-924d62bff616')
            .then(res => res.json())
            .then(
                result => {
                    this.setState({ isLoaded: true, scoreHome: result.homeScore, scoreVisitors: result.visitorsScore })
                    console.log('match: ', result)
                },
                error => {
                    this.setState({ isLoaded: false, error })
                    console.log('error: ', error)
                }
            )
    }

    render() {
        const { error, isLoaded, scoreHome, scoreVisitors } = this.state

        if (this.state.error) {
            return (
                <ErrorContainer>
                    <strong>Error:</strong> {error.message}
                </ErrorContainer>
            )
        } else if (!isLoaded) {
            return <LoadingDualRing></LoadingDualRing>
        } else {
            return (
                <section>
                    <ScoreContainer>
                        <TeamScore side={'HOME'} team={'SÃ£o Paulo'} score={scoreHome}></TeamScore>
                        <div>x</div>
                        <TeamScore side={'VISITORS'} team={'Santos'} score={scoreVisitors} reverse={true}></TeamScore>
                    </ScoreContainer>
                </section>
            )
        }
    }
}

export default ScoreBoard
