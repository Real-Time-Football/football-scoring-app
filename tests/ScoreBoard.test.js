import React from 'react'
import Enzyme from 'enzyme'
import ScoreBoard from '../src/components/ScoreBoard'

describe('Score Board', () => {
    let wrapper

    beforeEach(() => {
        global.fetch = jest.fn()

        global.fetch.mockImplementation(() =>
            Promise.resolve({
                status: 200,
                json: () =>
                    Promise.resolve({
                        homeScore: 2,
                        visitorsScore: 1,
                    }),
            })
        )

        wrapper = Enzyme.shallow(<ScoreBoard></ScoreBoard>)
    })

    afterEach(done => {
        wrapper.unmount()
        global.fetch.mockClear()
        done()
    })

    it('render a <section>', () => {
        expect(wrapper.type()).toBe('section')
    })

    it('update state `scoreHome` to 2', () => {
        expect(wrapper.state('scoreHome')).toBe(2)
    })

    it('update state `scoreVisitors` to 1', () => {
        expect(wrapper.state('scoreVisitors')).toBe(1)
    })

    it('renders the current score home in a TeamScore', () => {
        wrapper.setState({ scoreHome: 1 })
        console.log(wrapper)
        const teamHomeScore = wrapper
            .find('TeamScore')
            .at(0)
            .prop('score')
        expect(teamHomeScore).toBe(1)
    })

    it('renders the current score visitors in a TeamScore', () => {
        wrapper.setState({ scoreVisitors: 2 })
        const teamVisitorsScore = wrapper
            .find('TeamScore')
            .at(1)
            .prop('score')
        expect(teamVisitorsScore).toBe(2)
    })

    it('renders correctly', () => {
        wrapper = Enzyme.mount(<ScoreBoard></ScoreBoard>)
        expect(wrapper).toMatchSnapshot()
    })
})
