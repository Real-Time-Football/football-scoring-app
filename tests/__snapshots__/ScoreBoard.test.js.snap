// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score Board renders correctly 1`] = `
.c0 {
  display: inline-block;
  width: 80px;
  height: 80px;
}

.c0:after {
  content: ' ';
  display: block;
  width: 32px;
  height: 32px;
  margin: 10px;
  border-radius: 50%;
  border: 3px solid #6495ed;
  border-color: #6495ed transparent #6495ed transparent;
  -webkit-animation: eoUyJr 1.2s linear infinite;
  animation: eoUyJr 1.2s linear infinite;
}

<ScoreBoard>
  <ScoreBoard__LoadingDualRing>
    <div
      className="c0"
    />
  </ScoreBoard__LoadingDualRing>
</ScoreBoard>
`;
